### PR TITLE
MdePkg: Add Google Mock Library for SafeIntLib

### PR DESCRIPTION
**Pull request by TsunFeng**
# Description

MdePkg: Add Google Mock Library for PostCodeLib

<_For each item, place an "x" in between `[` and `]` if true. Example: `[x]` (you can also check items in GitHub UI)_>

<_Create the PR as a Draft PR if it is only created to run CI checks._>

<_Delete lines in \<\> tags before creating the PR._>

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Unit tests component can call these mock functions success

## Integration Instructions

N/A

### CODE REVIEW

**Review by lgao4 — APPROVED** [review]

File: MdePkg/Test/Mock/Include/GoogleTest/Library/MockSafeIntLib.h [hunk 1]
```diff
@@ -0,0 +1,1001 @@
+/** @file MockSafeIntLib.h
+  Google Test mocks for the SafeInt Library
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_INT_SAFE_LIB_H_
```
Comment (mdkinney) [inline MdePkg/Test/Mock/Include/GoogleTest/Library/MockSafeIntLib.h:8]: define name should be MOCK_SAFE_INT_LIB_H

File: MdePkg/Test/Mock/Include/GoogleTest/Library/MockSafeIntLib.h [hunk 2]
```diff
@@ -0,0 +1,1001 @@
+/** @file MockSafeIntLib.h
```
Comment (mdkinney) [inline MdePkg/Test/Mock/Include/GoogleTest/Library/MockSafeIntLib.h:1]: edk2 repo does not put the file name in doxygen @file statements.  Removes the need to update that content if file is renamed.

File: MdePkg/Test/Mock/Library/GoogleTest/MockSafeIntLib/MockSafeIntLib.cpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
+/** @file MockSafeIntLib.cpp
```
Comment (mdkinney) [inline MdePkg/Test/Mock/Library/GoogleTest/MockSafeIntLib/MockSafeIntLib.cpp:1]: edk2 repo does not put the file name in doxygen @file statements.  Removes the need to update that content if file is renamed.

File: MdePkg/Test/Mock/Include/GoogleTest/Library/MockSafeIntLib.h (hunk 2 above)
Comment (TsunFeng) [reply MdePkg/Test/Mock/Include/GoogleTest/Library/MockSafeIntLib.h]: Done

File: MdePkg/Test/Mock/Library/GoogleTest/MockSafeIntLib/MockSafeIntLib.cpp (hunk 3 above)
Comment (TsunFeng) [reply MdePkg/Test/Mock/Library/GoogleTest/MockSafeIntLib/MockSafeIntLib.cpp]: Done

File: MdePkg/Test/Mock/Library/GoogleTest/MockSafeIntLib/MockSafeIntLib.inf [hunk 4]
```diff
@@ -0,0 +1,33 @@
+## @file MockSafeIntLib.inf
```
[the remark this answers is not in the document]
Comment (TsunFeng) [reply MdePkg/Test/Mock/Library/GoogleTest/MockSafeIntLib/MockSafeIntLib.inf]: Done

**Comment by mdkinney** [comment]
All the updates look good.  Please combine back to one commit.

**Review by mdkinney — APPROVED** [review]

**Pull request merged**